src/middleware/insights.ts

```diff
@@ -95,6 +107,9 @@ aa('setUserToken', 'your-user-token');
       unsubscribe() {
         insightsClient('onUserTokenChange', undefined);
       },
+      sendEvent(event: InsightsEvent) {
```

Comment:
I think this should be added to InstantSearch directly

```
instantsearch.middlewareGrabBag.insights.SendEvent = ...
```

or even directly:

```
instantsearch.$insights.sendEvent
```

Maybe that's less easily typeable though

Reply:
I've changed it to `this._insightsMiddleware.sendEvent()`

Reply:
This can be a function in scope of the middleware, which is being called by by a function which is added to the instance in subscribe:

```
// pseudo-code
insights() {
  const sendEvent = ...
  return {
    subscribe() {
      instantsearch.sendInsightsEvent = sendEvent
```

instantsearch can then either have a noop sendEvent by default, or the widgets can check if sendEvent exists first

Reply:
I love this idea. It makes so many verbose things disappear. https://github.com/algolia/instantsearch.js/pull/4456/commits/19751711777f8621e483d5d4a33ada728f938ae6